kotlinx-coroutines-core/jvm/test/sync/MutexStressTest.kt

```diff
@@ -26,4 +26,34 @@ class MutexStressTest : TestBase() {
         jobs.forEach { it.join() }
         assertEquals(n * k, shared)
     }
+
+    @Test
+    fun stressUnlockCancelRace() = runTest {
+        val n = 10_000 * stressTestMultiplier
+        val mutex = Mutex(true) // create a locked mutex
+        newSingleThreadContext("SemaphoreStressTest").use { pool ->
+            repeat (n) {
+                // Initially, we hold the lock and no one else can `lock`,
+                // otherwise it's a bug.
+                assertTrue(mutex.isLocked)
+                var job1EnteredCriticalSection = false
+                val job1 = launch(start = CoroutineStart.UNDISPATCHED) {
+                    mutex.lock()
+                    job1EnteredCriticalSection = true
```

Comment:
Replacing it with
```
mutex.onLock {
    job1EnteredCriticalSection = true
    mutex.unlock()
}
```

triggers test failure

Reply:
Fixed